testament/testament.nim

```diff
@@ -1001,7 +1026,13 @@ proc codegenCheck(
   ) =
   ## Check for any codegen mismatches in file generated from `test` run.
   ## Only file that was immediately generated is tested.
+  # REFACTOR convert into procedure that creates `CodegenCheckResult`
```

Comment:
`CodegenCheckResult` or some other kind of report object -- anything aside from direct mutation of the specification